api/v1beta2/network_types.go

```diff
@@ -435,10 +435,13 @@ func (s Subnets) IDs() []string {
 }
 
 // FindByID returns a single subnet matching the given id or nil.
-func (s Subnets) FindByID(id string) *SubnetSpec {
-	for _, x := range s {
+//
+// The returned pointer can be used to write back into the original slice.
+func (s *Subnets) FindByID(id string) *SubnetSpec {
```

Comment:
Changing the receiver to a pointer is causing a apidiff failure:
```
 sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2
  Incompatible changes:
  - Subnets.FindByID: removed
  - Subnets.FindEqual: removed 
```

Normally we ignore these errors but as this is inside the `api` folder then i don't think we can ignore this, sorry. 

Would you be able to change this and **FindEqual** back?

Reply:
Makes sense. I've marked the existing ones as deprecated in favor of the (now renamed) `Find*Ptr` variants.

Also rebased and adapted the previous state (commit 987df2437150176c2c671908474ae9565f2f6d60) onto `main` to ensure that tests still work.

Reply:
Thanks for changing this.